### PR TITLE
Feat/tm 5730 pricebook

### PR DESCRIPTION
**Pull request by barbarafarias**

### CODE REVIEW

File: lib/v1/products/index.js [hunk 1]
```diff
@@ -102,6 +103,35 @@ module.exports.prices = {
...
+    },
+    entry: {
+      create: {
+        request: {
+          $id: 'https://schemas.tillhub.com/v1/products.prices.book.entry.create.request.schema.json',
+          $schema: 'http://json-schema.org/draft-07/schema#',
+          additionalProperties: false,
+          type: 'object',
+          required: [
+            'price_book_id',
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/index.js]: `price_book`

File: lib/v1/products/index.js [hunk 2]
```diff
@@ -102,6 +103,35 @@ module.exports.prices = {
...
+            'product_id'
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/index.js]: `product`

File: lib/v1/products/prices/book/base.js [hunk 3]
```diff
@@ -1,103 +1,33 @@
 const { oneOf } = require('../../../../helpers/payload-or-null')
 
 module.exports = {
-  product: {
-    description: 'The reference to the product',
+  id: {
+    description: 'The Price Book id',
     type: 'string',
     format: 'uuid'
   },
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/prices/book/base.js]: The caller should not be able to send the ID. Just remove it

File: lib/v1/products/prices/book/base.js [hunk 4]
```diff
@@ -1,103 +1,33 @@
...
-  active: {
-    type: 'boolean'
+  constraints: {
+    ...oneOf(require('./constraints'))
   },
-  deleted: {
-    type: 'boolean'
-  },
-  locations: oneOf({
+  policies: {
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/prices/book/base.js]: Let's remove policies as this policies are their own object reference to a resource.

File: lib/v1/products/prices/book/entry/base.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
+const { oneOf } = require('../../../../../helpers/payload-or-null')
+
+module.exports = {
+  price_book_id: {
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/prices/book/entry/base.js]: `price_book`

File: lib/v1/products/prices/book/entry/base.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    description: 'The reference to the product',
+    type: 'string',
+    format: 'uuid'
+  },
+  product_id: {
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/prices/book/entry/base.js]: `product`

File: lib/v1/products/prices/book/entry/base.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  rate: {
+    description: 'Relative value of this product based on any other applicable price.',
+    ...oneOf({
+      type: 'number',
+      minimum: 0,
+      maximum: 9,
+      multipleOf: 0.001
+    })
+  },
+  margin_rate: {
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/prices/book/entry/base.js]: margin appears to be computed, last time I checked. Let's remove it

File: lib/v1/products/prices/book/entry/base.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  net_book_price: {
+    ...oneOf({
+      example: '27633.02',
+      type: 'number',
+      minimum: 0,
+      maximum: 1000000,
+      multipleOf: 0.01
+    })
+  },
+  gross_book_price: {
```
Comment (eljefedelrodeodeljefe) [inline lib/v1/products/prices/book/entry/base.js]: Please use the canonical amount namings `amount_gross` etc.

**Comment by eljefedelrodeodeljefe** [comment]
Please squash

File: lib/v1/products/prices/book/base.js (hunk 3 above)
Comment (barbarafarias) [reply lib/v1/products/prices/book/base.js]: @eljefedelrodeodeljefe But in the frontend there are 2 fields when creating a price book: name and id.

There should be at least a custom_id property or something, right?

Comment (eljefedelrodeodeljefe) [reply lib/v1/products/prices/book/base.js]: custom_id

**Pull request merged**